src/util/bip32.rs

```diff
@@ -187,6 +207,24 @@ impl serde::Serialize for ChildNumber {
     }
 }
 
+/// Parse a BIP32 derivation path.
+pub fn parse_derivation_path(path: &str) -> Result<Vec<ChildNumber>, Error> {
+    let mut parts = path.split("/");
+    // First parts must be `m`.
+    if parts.next().unwrap() != "m" {
+        return Err(Error::InvalidDerivationPathFormat);
+    }
+
+    let mut childnumbers = Vec::new();
```

Comment:
This and the following lines in this block can be replaced with simply:

```rust
    parts.map(|p| p.parse()).collect()
```

See the [last example here](https://doc.rust-lang.org/std/iter/trait.Iterator.html#examples-24) detailing what happens when you call `.collect()` on a list of `Result`s for why this works.

Reply:
You could further replace `|p| p.parse()` with `str::parse`, i.e. `parts.map(str::parse).collect()`

Reply:
Yeah so I just remember that I tried that, but then you can't use the `try!` thing. Because `parse` returns a `Result`. So you'd need a method that converts an `Iterator<Result<T,E>>` into an `Result<Iterator<T>, E>`.

Reply:
Did this. Didn't know you could `collect` to a Result. Very cool!